fenix/app/src/main/java/org/mozilla/fenix/compose/SwitchToggle.kt

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Switch
+import androidx.compose.material.SwitchDefaults
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.compositeOver
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+
+private const val DISABLED_ALPHA = 0.5f
+
+/**
+ * UI for a switch that can be on or off.
+ *
+ * @param checked whether or not this component is checked.
+ * @param onCheckedChange invoked when Switch is being clicked, therefore the change of checked
+ * state is requested.
+ * @param modifier Modifier to be applied to the switch layout
+ * @param enabled whether the component is enabled or grayed out
+ */
+@Composable
+fun SwitchToggle(
+    checked: Boolean,
+    onCheckedChange: ((Boolean) -> Unit),
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+) {
+    Switch(
```

Comment:
```suggestion
fun Switch(
    checked: Boolean,
    onCheckedChange: ((Boolean) -> Unit),
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
) {
    androidx.compose.material.Switch(
```

this mirrors what we've done in `Divider`, `TextButton`, `Button`, and a couple other places

Reply:
+1 to the using the `Switch` naming since this will mirror naming in the figma component.